### PR TITLE
Feat/delete schema

### LINKED ISSUE #36: Delete Schema

### PR DESCRIPTION
**Pull request by jamieknight-db**
https://github.com/user-attachments/assets/fdbb8864-e157-4b19-b9d6-6082abec24ab

Closes #36

### CODE REVIEW

**Comment by jamieknight-db** [comment]
@xx-db I'm tagging you in a comment in hopes that it lets me add you as a reviewer 🙏

File: src/hooks/schemas.tsx [hunk 1]
```diff
@@ -90,3 +90,39 @@ export function useCreateSchema({
...
+export interface DeleteSchemaMutationParams
+  extends Pick<SchemaInterface, 'catalog_name' | 'name'> {}
+
+interface DeleteSchemaParams {
+  onSuccessCallback?: () => void;
+  catalog: string;
+}
+
+export function useDeleteSchema({
+  onSuccessCallback,
```
Comment (yc-shawn) [inline src/hooks/schemas.tsx:103]: Sorry I just captured this.
You don't need to do create a new prop here to handle on success.

You can just simply do onSuccess in the caller
```tsx
mutation.mutate(values, {
  onSuccess: () => {}
});
```
Can we have follow-up PRs to remove this pattern on other deletion hooks?

**Review by yc-shawn — APPROVED** [review]

File: src/hooks/schemas.tsx (hunk 1 above)
Comment (jamieknight-db) [reply src/hooks/schemas.tsx:103]: Do you mean to move the `invalidateQueries` to the caller as well?

Comment (yc-shawn) [reply src/hooks/schemas.tsx:103]: Similar example: https://github.com/TanStack/query/blob/1fc6124bfd22a26095ab1fca20caa39b4036fd86/examples/react/auto-refetching/src/pages/index.tsx#L39

Comment (jamieknight-db) [reply src/hooks/schemas.tsx:103]: Ok got it, but we don't need to pass in `onSuccessCallback`. I will do a follow-on PR to remove all those. Thanks @yc-shawn 👍

**Review by xx-db — APPROVED** [review]
lgtm 💪

**Review by Romanize — APPROVED** [review]

**Pull request merged**